### PR TITLE
WasmThemis example project with webpack

### PR DESCRIPTION
**Pull request by ilammy**
Finally my dream is coming true: interactive example of Themis right in the browser. It's a bit sad that I have to work for my own dreams. Oh well, doing what I can with what I got.

Try it out by running:

```
cd docs/examples/wasm/webpack
npm install
npm run start
```

Or if you're lazy – I know you are – [check out the hosted version](https://wasm-themis-example.ilammy.net/).

Again, @maxammann, if you would be so kind to give this a look – I'd be grateful.

### Known issues

- **Resolved:** ~~`wasm-themis@0.13.11` is not published.~~ (Now it is.)

  Since the currently published `0.13.1` is not working in browsers, I'm using my preview package. It would be nice to get WasmThemis patch release into npm for the example to be “official”.

- Page load time is abysmal.

  webpack correctly warns you that putting 1.8 MB of JS and 1.6 MB of WASM is probably not the greatest idea for quick page loads. It takes a couple of seconds for all this stuff to download and initialize. I have not optimized it whatsoever.

- Secure Cell performance with passpharases.

  While symmetric key encryption works reasonably well for interactive use, passphrase encryption is slow—well, it's designed to be. Since all of this is running on the single UI thread, the interface not very responsive as WasmThemis just can't process input as fast as you type it.

- Something seems to be broken with `mode: 'production'`

  Building the site in production mode causes `TypeErrors` due to `null` values popping up where they are not expected. No idea what causes this, the code is all minified and undebuggable.

- No tests on CI whatsoever. And code coverage is low.

  I mean, it would be trivial to test *building* the example. But this does not guarantee that it actually works in the browser. Maybe later.

  Similarly, only Secure Cell is shown because I have only so much life to waste on this. Maybe later.

You will also have to excuse complete lack of JS frameworks and my “impressive” web design skills. Patches are welcome.

## Checklist

- [x] ~~Change is covered by automated tests~~ (no, it is not; not in the slightest)
- [x] Benchmark results are attached (watch for yourself)
- [x] The [coding guidelines] are followed
- [x] Public API has proper documentation
- [x] Example projects and code samples are up-to-date
- [x] ~~Changelog is updated~~ (don't need, I guess?)

[coding guidelines]: https://github.com/cossacklabs/themis/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: docs/examples/wasm/webpack/src/base64.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+function uint6ToB64(c) {
+    return c < 26 ? c + 65
+         : c < 52 ? c + 71
+         : c < 62 ? c - 4
+         : c === 62 ? 43
+         : c === 63 ? 47
+         : 65
+}
+
+export function Base64ToBytes(sBase64, nBlocksSize) {
```
Comment (Lagovas) [inline docs/examples/wasm/webpack/src/base64.js:23]: i'm so glad that you didn't copy-pasted code from mozilla's examples with their naming and changed it to readable function names)

File: docs/examples/wasm/webpack/src/secure-cell.js [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+        try {
+            const decodedKey = Base64ToBytes(symmetricKeyInput.value)
+            const encodedContext = StringToUTF8Array(contextArea.value)
+            const decodedCiphertext = Base64ToBytes(ciphertextSealArea.value)
+
+            const cell = themis.SecureCellSeal.withKey(decodedKey)
+            const plaintext = cell.decrypt(decodedCiphertext, encodedContext)
+
+            plaintextArea.value = UTF8ArrayToString(plaintext)
+            plaintextLength.textContent = `${plaintext.length} bytes`
```
Comment (Lagovas) [inline docs/examples/wasm/webpack/src/secure-cell.js:107]: wow how JS can do ) situation when hasn't used js several years and meet new features

File: docs/examples/wasm/webpack/src/secure-cell.js [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+    function encryptWithPassphrase() {
+        encryptSealPassphrase()
+    }
+
+    let encrypt = encryptWithKey
+    let decryptSeal = decryptSealWithKey
+
+    let update = function(){}
+
+    generateButton.addEventListener('click', function() {
```
Comment (Lagovas) [inline docs/examples/wasm/webpack/src/secure-cell.js:295]: so pretty code structure and readable variable names... so easy to read and understand, thank you. cool example project

**Review by Lagovas — APPROVED** [review]

**Comment by vixentael** [comment]
> Since the currently published 0.13.1 is not working in browsers

working on it...

File: docs/examples/wasm/webpack/webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        new HtmlPlugin({
+            filename: 'index.html',
+            template: 'src/index.html',
+        }),
+        new SRIPlugin({
+            hashFuncNames: ['sha256'],
+        }),
+    ],
+    // libthemis.wasm and it's JS bundle is over 1 megabyte.
+    // It's okay. Please be quiet, webpack.
```
Comment (vixentael) [inline docs/examples/wasm/webpack/webpack.config.js:49]: hah

File: docs/examples/wasm/webpack/webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+            util: false,
+        }
+    },
+    // Bundle libthemis.wasm with the script.
+    // Compute SRI hash sum of the script and inject it into HTML.
+    plugins: [
+        new CopyPlugin({
+            patterns: [
+                // FIXME: use "wasm-themis" instead of "@ilammy/wasm-themis-preview"
+                { from: 'node_modules/@ilammy/wasm-themis-preview/src/libthemis.wasm',
```
Comment (vixentael) [inline docs/examples/wasm/webpack/webpack.config.js:35]: 👀

**Review by vixentael — APPROVED** [review]
That's look fantastic, thank you @ilammy!
 
Screenshot 2021-07-06 at 01 35 31

I'm working on getting wasm-themis-0.13.11 published

**Pull request merged**